tools/ci_build/github/azure-pipelines/mac-ios-ci-pipeline.yml

```diff
@@ -63,34 +63,33 @@ jobs:
           --path_to_protoc_exe $(Build.BinariesDirectory)/protobuf_install/bin/protoc \
           --parallel
       displayName: (CoreML EP) Build onnxruntime for iOS x86_64 and run tests using simulator
+- ${{ or(if eq(parameters.BuildXnnpack, 'true'), if eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
```

Comment:
We won't know if it works until you merge this PR.

Reply:
Now, we know,
- 1. `parameters.BuildXnnpack` it works, I have tested it.
- 2. when `variables['Build.SourceBranch']` is not equal to `'refs/heads/main'`, it doesn't build

Reply:
I can test it by merging it into another branch, and have the condition of comparing it to the target branch.